csvcubed/csvcubed/utils/pandas.py

```diff
@@ -122,3 +135,23 @@ def coalesce_on_uri_safe(data: PandasDataTypes) -> PandasDataTypes:
                     cat_map[f] = sorted(v)[0]
 
             return data.map(cat_map).astype("category")
+
+
+def _data_is_string_type(data: PandasDataTypes) -> bool:
+    if data is None:
+        return False
+
+    if data.dtype == "category":
+        return is_string_dtype(data.cat.categories)
+
+    return is_string_dtype(data)
+
+
+def _get_unique_values(data: PandasDataTypes) -> Iterable[str]:
+    if data is None:
+        return []
+
+    if data.dtype == "category":
```

Comment:
This won't work because you can't get dtype from a pd.DataFrame, it will need to be coalesced to a Series first.

Reply:
noooooooo

Reply:
I've made the input param type more specific to avoid this problem.